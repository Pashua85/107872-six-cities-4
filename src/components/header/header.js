import React from 'react';
import PropTypes from 'prop-types';
import {Link} from 'react-router-dom';
import {connect} from 'react-redux';
import {getAuthStatus} from '../../store/reducers/authStatusReducer/selectors';
import {getUser} from '../../store/reducers/userReducer/selectors';
import {AUTH_STATUS} from '../../store/reducers/authStatusReducer/authStatusReducer';

const Header = (props) => {
  const {authStatus, user} = props;

  return (
    <header className="header">
      <div className="container">
        <div className="header__wrapper">
          <div className="header__left">
            <Link className="header__logo-link" to="/">
              <img className="header__logo" src="/img/logo.svg" alt="6 cities logo" width="81" height="41" />
            </Link>
          </div>
          <nav className="header__nav">
            <ul className="header__nav-list">
              <li className="header__nav-item user">
                {
                  authStatus === AUTH_STATUS.NO_AUTH &&
                  (
                    <Link className="header__nav-link header__nav-link--profile" to="/login">
                      <div className="header__avatar-wrapper user__avatar-wrapper">
                      </div>
                      <span className="header__login">Sign in</span>
                    </Link>
                  )
                }
                {
                  authStatus === AUTH_STATUS.AUTH && user !== null &&
                  (
                    <Link className="header__nav-link header__nav-link--profile" to="/favorites">
                      <div className="header__avatar-wrapper user__avatar-wrapper">
                      </div>
                      <span className="header__user-name user__name">{user.email}</span>
                    </Link>
                  )
                }
              </li>
            </ul>
          </nav>
        </div>
      </div>
    </header>
  );
};

Header.propTypes = {
  authStatus: PropTypes.string.isRequired,
  user: PropTypes.oneOfType([PropTypes.oneOf([null]), PropTypes.object])
};

const mapStateToProps = (state) => ({
  authStatus: getAuthStatus(state),
  user: getUser(state)
});

export default connect(mapStateToProps)(Header);
export {Header};
