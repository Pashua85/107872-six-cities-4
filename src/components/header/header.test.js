import React from 'react';
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {Header} from './header';

Enzyme.configure({
  adapter: new Adapter()
});

describe(`Header`, () => {
  it(`should render correctly for guest`, () => {
    const wrapper = shallow(
        <Header authStatus={`NO_AUTH`} user={{email: `email`}} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});


