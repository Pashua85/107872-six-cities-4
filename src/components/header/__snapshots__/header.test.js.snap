// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render correctly for guest 1`] = `ShallowWrapper {}`;
